README.md

```diff
@@ -2,30 +2,166 @@
-the tests and other means of validating successful completion of deliverables.
+```java
+public abstract class Pet {
 
-## Conclusion
+    private String name;
+    private boolean likesBaths;
 
-A short one or two paragraph summary of the contents of the lessons, recapping
-the learning goals.
+    public Pet(String name, boolean likesBaths) {
+        this.name = name;
+        this.likesBaths = likesBaths;
+    }
 
-## Resources
+    public String getName() { return name; }
+    public boolean likesBaths() { return likesBaths; }
+
+    public abstract void giveTreat();
+    public abstract void giveBath();
+
+}
+```
+
+
+
+## Task #1 - Implement the `Dog` subclass
+
+Create a new class named `Dog` that extends `Pet`.
+
+- `Dog` should have an additional instance variable `tailWagging` of type `boolean`.
+- The `Dog` constructor should take 3 parameters: `name`, `likesBaths`, and `tailWagging`.
+    - Call the superclass constructor to initialize the inherited `name` and `likesBaths` fields.
+    - Initialize the `tailWagging` field.
+- Add an accessor method `isTailWagging` to return the value of the `tailWagging` field.
+- Override `giveTreat()` to assign `tailWagging` to `true`.
+- Override `giveBath()` to assign `tailWagging` based on the value of the inherited field `likesBaths`.
+    - NOTE: The inherited field is private, so call the public accessor method of the same name.
+
+Create a new Junit class in the `test/java` folder to test the `Dog` class:
+
+```java
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class DogTest {
+
+    private Dog dog1, dog2;
+
+    @BeforeEach
+    void setUp() {
+        dog1 = new Dog("Rex", true, false);
+        dog2 = new Dog("Rover", false, true);
+    }
+
+    @Test
+    void constructor() {
+        assertEquals("Rex", dog1.getName());
+        assertTrue(dog1.likesBaths());
+        assertFalse(dog1.isWaggingTail());
+
+        assertEquals("Rover", dog2.getName());
+        assertFalse(dog2.likesBaths());
+        assertTrue(dog2.isWaggingTail());
+
+    }
+
+    @Test
+    void giveTreat() {
+        dog1.giveTreat();
+        assertTrue(dog1.isWaggingTail());
+
+        dog2.giveTreat();
+        assertTrue(dog2.isWaggingTail());
+    }
+
+    @Test
+    void giveBath() {
+        dog1.giveBath();
+        assertTrue(dog1.isWaggingTail());
+
+        dog2.giveBath();
+        assertFalse(dog2.isWaggingTail());
+    }
+}
+```
+
+
+## Task #2 - Implement the `Cat` subclass
+
+Create a new class named `Cat` that extends `Pet`.
+
+- `Cat` should have an additional instance variable `purring` of type `boolean`.
+- We will assume all cats **do not** like baths, so the `Cat` constructor only takes
+  2 parameters: `name` and `purring`.
+    - Call the superclass constructor to initialize the inherited `name` and `likesBaths` fields.
+      Remember, cats do not like baths!
```

Comment:
Fun fact about my cat, Riley! She loves water! She takes a shower with me almost every day and constantly asks the water to be turned on in the sink so she can play with it haha! 

I recognize her as the outlier of cats, as my other cat is normal and does not like baths - but does dramatically tolerate them.

Reply:
My son's cat actually loves water too and likes to sit in the kitchen sink!

Reply:
love it!